node-services/user-service/controllers/userDb.js

```diff
@@ -0,0 +1,76 @@
+const db = require('../dbInstance');
+const bcrypt = require('bcrypt');
+const { validateEmail, validateUserParameter } = require('./validateParameters');
+
+//Response codes for signup
+// 0 -> user already exists
+// 1 -> user created successfully 
+
+
+//Response codes for login
+// -1 -> Wrong password
+//  0 -> User doesn't exists
+//  1 -> User login successful
+
+const dbOperations = {
+  signup: async (User) => {
+    if (!validateEmail(User.email)) {
+      throw new Error("Invalid email format");
+    }
+    
+    if (!validateUserParameter(User.name) || !validateUserParameter(User.password) ||
+        !validateUserParameter(User.age) || !validateUserParameter(User.gender)
+    ) {
+      throw new Error("Parameters can't be empty");
```

Comment:
this is where validate method throwing error becomes clean (since you are anyway throwing error). code also becomes cleaner like you just called methods without any if checks. Optional btw